environs/gui/dashboard.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package gui
+
+import (
+	"archive/tar"
+	"compress/bzip2"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/version"
+)
+
+// DashboardArchiveVersion retrieves the Dashboard version
+// from the given tar.bz2 archive reader.
+func DashboardArchiveVersion(r io.Reader) (version.Number, error) {
```

Comment:
moved from elsewhere so this code can be shared, it was copied in a couple of places